doc/specifications/proposed/replicator/architecture.rst

```diff
@@ -0,0 +1,142 @@
+==========
+Replicator
+==========
+
+This document describes a common synchronization mechanism between two systems,
+which can cope with write failures, network downtimes and other types of
+partitions. The approach only works with a single write master (Single Source
+of Truth) and any number of read slaves.
+
+In our case we want to use this approach to synchronize the content database
+with a search index.
+
+General Workings
+================
+
+The general scheme for synchronizations is fairly simple:
+
+.. image:: synchronize.png
+
+It generally can be described as a three step process:
+
+1) Fetch last revision from target
```

Comment:
The most common usecase would be to seed the target using a copy of the master database. I think this should be mentioned in the spec